packages/cli/core/src/commands/ui/deploy.ts

```diff
@@ -0,0 +1,184 @@
+    patisModuleh: {type: 'boolean'},
+  },
+};
+
+const DeployConfigSchema: Schema = {
+  type: 'object',
+  required: ['name', 'dir', 'htmlEntryFile'],
+  properties: {
+    name: {type: 'string'},
+    dir: {type: 'string'},
+    htmlEntryFile: FileInputSchema,
+    javascriptEntryFiles: {type: 'array', items: JavaScriptFileInputSchema},
+    javascriptUrls: {type: 'array', items: JavaScriptFileInputSchema},
+    cssEntryFiles: {type: 'array', items: FileInputSchema},
+    cssUrls: {type: 'array', items: FileInputSchema},
+  },
+};
+
+export default class Deploy extends CLICommand {
+  public static hidden = true; /* TODO: uncomment */
+  public static flags = {
+    pageId: Flags.string({
+      char: 'p',
+      description: 'The existing ID of the Hosted Page to be updated.',
+      helpValue: '7944ff4a-9943-4999-a3f6-3e81a7f6fb0a',
+      required: false,
+    }),
+    config: Flags.string({
+      char: 'c',
+      description: 'The path to the deploy JSON configuration.',
+      helpValue: 'coveo.deploy.json',
+      default: 'coveo.deploy.json',
+      required: false,
+    }),
+  };
+
+  @Trackable()
+  @Preconditions(
+    IsAuthenticated([AuthenticationType.OAuth]),
+    HasNecessaryCoveoPrivileges(createSearchPagesPrivilege)
+  )
+  public async run() {
+    const {flags} = await this.parse(Deploy);
+    const deployConfigPath = flags.config;
+    const deployConfig: DeployConfig =
+      readJsonSync(deployConfigPath, {throws: false}) || {};
+    this.validateDeployConfigJson(deployConfigPath, deployConfig);
+    const hostedPage = this.buildHostedPage(deployConfig);
+
+    if (flags.pageId) {
+      this.updatePage({...hostedPage, id: flags.pageId});
+      return;
+    }
+
+    try {
+      await this.createPage(hostedPage);
+    } catch (error) {
+      /**
+       * TODO: handle Platform Client 400 error e.g.:
+       * {
+       *  statusCode: 400,
+       *  message: "A configuration named 'my page' already exist.",
+       *  type: 'DuplicateConfigurationException'
+       * }
+       */
+      console.error(error);
+      super.catch(error);
+    }
+  }
+
+  private validateDeployConfigJson(
+    deployConfigPath: string,
+    config: DeployConfig
+  ) {
+    const validation = validate(config, DeployConfigSchema);
+    if (!validation.valid) {
+      throw new DeployConfigError(deployConfigPath, validation.errors);
+    }
+  }
+
+  private buildHostedPage({
+    dir,
+    htmlEntryFile,
+    name,
+    cssEntryFiles = [],
+    cssUrls = [],
+    javascriptEntryFiles = [],
+    javascriptUrls = [],
+  }: DeployConfig): New<HostedPage> {
+    ensureDirSync(dir);
+
+    return {
+      name,
+      html: readFileSync(join(dir, htmlEntryFile.path), 'utf-8'),
+      javascript: [
+        ...javascriptEntryFiles.map(({path, isModule}) => ({
+          inlineContent: readFileSync(join(dir, path), 'utf-8'),
+          isModule,
+        })),
+        ...javascriptUrls.map(({path, isModule}) => ({
+          inlineContent: path,
```

Comment:
Is this meant to be `url`?
```suggestion
          url: path,
```